hail/src/main/scala/is/hail/expr/ir/streams/EmitStream.scala

```diff
@@ -2761,6 +2765,85 @@ object EmitStream {
           SStreamValue(producer)
         }
 
+      case StreamLocalLDPrune(a, r2Threshold, winSize, maxQueueSize, nSamples) =>
+        produce(a, cb)
+          .map(cb) { case childStream: SStreamValue =>
+            val childProducer = childStream.producer
+            val queueSize = cb.newField[Int]("max_queue_size")
+            val queue = cb.newField[util.ArrayDeque[BitPackedVector]]("queue")
+            val threshold = cb.newField[Double]("r2_threshold")
+            val windowSize = cb.newField[Int]("window_size")
+            val builder = cb.newField[BitPackedVectorBuilder]("vector_builder")
+            val elementType = typeWithReq.canonicalEmitType.st.asInstanceOf[SStream].elementEmitType
+            val elementField = mb.newEmitField(elementType)
+
+            val producer = new StreamProducer {
+              val length: Option[EmitCodeBuilder => Code[Int]] = None
+
+              val elementRegion: Settable[Region] = childProducer.elementRegion
+
+              val element: EmitCode = elementField
+
+              val requiresMemoryManagementPerElement: Boolean = childProducer.requiresMemoryManagementPerElement
+
+              def initialize(cb: EmitCodeBuilder): Unit = {
+                cb.assign(queueSize, emit(maxQueueSize, cb).get(cb).asInt32.value)
+                cb.assign(queue, Code.newInstance[util.ArrayDeque[BitPackedVector], Int](queueSize))
+                cb.assign(threshold, emit(r2Threshold, cb).get(cb).asFloat64.value)
+                cb.assign(windowSize, emit(winSize, cb).get(cb).asInt32.value)
+                cb.assign(builder, Code.newInstance[BitPackedVectorBuilder, Int](emit(nSamples, cb).get(cb).asInt32.value))
+                childProducer.initialize(cb)
+              }
+
+              val LproduceElement: CodeLabel = mb.defineAndImplementLabel { cb =>
+                val Lpruned = CodeLabel()
+
+                cb.goto(childProducer.LproduceElement)
+                cb.define(childProducer.LproduceElementDone)
+
+                childProducer.element.toI(cb).consume(cb,
+                  cb.goto(Lpruned),
+                  { case sc: SBaseStructValue =>
+                    val locus = sc.loadField(cb, "locus").get(cb).asLocus
+                    val locusObj = locus.getLocusObj(cb)
+                    val alleles = sc.loadField(cb, "alleles").get(cb).castTo(cb, elementRegion, SJavaArrayString(true), false).asInstanceOf[SJavaArrayStringValue]
```

Comment:
after tracking through the BitPackedVectorBlah classes, I think there's actually no reason that the alleles need to be passed through here. Can we strip that out?

Reply:
You're correct. However, we do still need the parameter to handle the unstaged code (for now).

Reply:
Can we just delete the unstaged one? You can define the lowering to TableToTableFunction, then implement the execute with staged boundaries.

Reply:
finally did this, we'll see if tests pass, if they do, this is ready to go in.

Reply:
great!